Examples/Algorithms/TrackFinding/src/TrackParamsEstimationAlgorithm.cpp

```diff
@@ -0,0 +1,117 @@
+// This file is part of the Acts project.
+//
+// Copyright (C) 2021 CERN for the benefit of the Acts project
+//
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at http://mozilla.org/MPL/2.0/.
+
+#include "ActsExamples/TrackFinding/TrackParamsEstimationAlgorithm.hpp"
+
+#include "Acts/EventData/TrackParameters.hpp"
+#include "Acts/Seeding/EstimateTrackParamsFromSeed.hpp"
+#include "ActsExamples/EventData/IndexSourceLink.hpp"
+#include "ActsExamples/EventData/SimSpacePoint.hpp"
+#include "ActsExamples/EventData/Track.hpp"
+#include "ActsExamples/Framework/WhiteBoard.hpp"
+
+#include <map>
+#include <stdexcept>
+
+ActsExamples::TrackParamsEstimationAlgorithm::TrackParamsEstimationAlgorithm(
+    ActsExamples::TrackParamsEstimationAlgorithm::Config cfg,
+    Acts::Logging::Level lvl)
+    : ActsExamples::BareAlgorithm("TrackParamsEstimationAlgorithm", lvl),
+      m_cfg(std::move(cfg)) {
+  if (m_cfg.inputSeeds.empty()) {
+    throw std::invalid_argument("Missing input seeds collection");
+  }
+  if (m_cfg.inputSourceLinks.empty()) {
+    throw std::invalid_argument("Missing input source links collection");
+  }
+  if (m_cfg.outputTrackParameters.empty()) {
+    throw std::invalid_argument("Missing output track parameters collection");
+  }
+  if (m_cfg.outputTrackParamsSeedMap.empty()) {
+    throw std::invalid_argument(
+        "Missing output trackparameters-to-seed collection");
+  }
+  if (not m_cfg.trackingGeometry) {
+    throw std::invalid_argument("Missing tracking geometry");
+  }
+}
+
+ActsExamples::ProcessCode ActsExamples::TrackParamsEstimationAlgorithm::execute(
+    const ActsExamples::AlgorithmContext& ctx) const {
+  using SeedContainer = std::vector<std::vector<Acts::Seed<SimSpacePoint>>>;
```

Comment:
Sounds good. Added.

Reply:
I understand the Seedcontainer looks like this due to what the seedfinder returns , which is basically an implementation detail of this algorithm (it can process each region in parallel). I'm not sure if we should couple the two algorithms this tightly, or if we should just say "seeds are stored in a single vector".

Reply:
I would prefer this as well. For the examples, it would be nice to actually store two different containers: a `ProtoTrackContainer` that just stores which hits were associated to a seed and a `TrackParametersContainer` for the extracted parameters. With the former, you can use the existing performance writers for track finding (seeding is just very coarse track finding) and the latter is the only thing that is needed e.g. for the CKF.

Reply:
@msmk0 , do you mean we use the `ProtoTrackContainer` in this example and elsewhere for storing the seed? I think this is a good idea.